Knuth_Morris_Pratt_Algorithm/KMP.c

```diff
@@ -0,0 +1,90 @@
+//Implementation of the Knuth Morris Pratt algorithm for pattern searching in C language
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<memory.h>
+
+char *parent = "acabacdabac";
```

Comment:
Remove these variables(you haven't used them) and if required, input them dynamically.